flytekit/models/sagemaker/parameter_ranges.py

```diff
@@ -199,3 +207,73 @@ def from_flyte_idl(cls, pb2_object: _idl_parameter_ranges.ParameterRanges):
                 converted[k] = CategoricalParameterRange.from_flyte_idl(v.categorical_parameter_range)
 
         return cls(parameter_range_map=converted,)
+
+
+class ParameterRangeOneOf(_common.FlyteIdlEntity):
+    def __init__(self, param: Union[IntegerParameterRange, ContinuousParameterRange, CategoricalParameterRange]):
+        """
+        Initializes a new ParameterRangeOneOf.
+        :param Union[IntegerParameterRange, ContinuousParameterRange, CategoricalParameterRange] param: One of the
+                supported parameter ranges.
+        """
+        self._integer_parameter_range = param if isinstance(param, IntegerParameterRange) else None
+        self._continuous_parameter_range = param if isinstance(param, ContinuousParameterRange) else None
+        self._categorical_parameter_range = param if isinstance(param, CategoricalParameterRange) else None
+
+    @property
+    def integer_parameter_range(self) -> Optional[IntegerParameterRange]:
+        """
+        Retrieves the integer parameter range if one is set. None otherwise.
+        :rtype: Optional[IntegerParameterRange]
+        """
+        if self._integer_parameter_range:
+            return self._integer_parameter_range
+
+        return None
+
+    @property
+    def continuous_parameter_range(self) -> Optional[ContinuousParameterRange]:
+        """
+        Retrieves the continuous parameter range if one is set. None otherwise.
+        :rtype: Optional[ContinuousParameterRange]
+        """
+        if self._continuous_parameter_range:
+            return self._continuous_parameter_range
+
+        return None
+
+    @property
+    def categorical_parameter_range(self) -> Optional[CategoricalParameterRange]:
+        """
+        Retrieves the categorical parameter range if one is set. None otherwise.
+        :rtype: Optional[CategoricalParameterRange]
+        """
+        if self._categorical_parameter_range:
+            return self._categorical_parameter_range
+
+        return None
+
+    def to_flyte_idl(self) -> _idl_parameter_ranges.ParameterRangeOneOf:
+        return _idl_parameter_ranges.ParameterRangeOneOf(
+            integer_parameter_range=self.integer_parameter_range.to_flyte_idl()
+            if self.integer_parameter_range
+            else None,
+            continuous_parameter_range=self.continuous_parameter_range.to_flyte_idl()
+            if self.continuous_parameter_range
+            else None,
+            categorical_parameter_range=self.categorical_parameter_range.to_flyte_idl()
+            if self.categorical_parameter_range
+            else None,
+        )
+
+    @classmethod
+    def from_flyte_idl(cls, pb_object: _idl_parameter_ranges.ParameterRangeOneOf):
```

Comment:
Oh I see. I was missing this intermediate layer. That's why my `ParameterRanges.from_flyte_idl()` didn't work as expected.